### DISCUSSION TITLE
Docker-compose: services with volume "." are always empty

### DISCUSSION THREAD

**Discussion by jehon** [devcontainers/spec#182] [Q&A]
In docker-compose, I have multiple services with "volume" mapped to "."

In the "devcontainer", this is mapped to a "volume", and it works fine.

But in other services, the "." is not handled.

How to map the "." to the volume?

It could be great to have a "variable" holding either "." or the volume name, but how to construct that?

**Comment by jehon** [comment]
I found some stuff:

- https://stackoverflow.com/q/65343518/1954789: how to map to LOCAL_WORKSPACE_FOLDER
But that requires a local checkout before launching the devcontainer

- https://github.com/h4l/dev-container-docker-compose-volume-or-bind
It looks for mounted volumes inside the devcontainer, and extract the name of the volume, then it use it for other mounts. But that a bit complex.

**Reply by chrmarti** [reply]
Came here to suggest using `"initializeCommand"` like https://github.com/h4l/dev-container-docker-compose-volume-or-bind does. Instead of the longish shell script you could use default values as suggested in https://stackoverflow.com/a/69830544/7857763 for the bind mounts and have a short script called from `"initializeCommand"` that writes an `.env` file for the volume case. For the script to check if it is run in the volume case we don't have a flag at the moment, maybe use `workspace_volume_name()` from the script in https://github.com/h4l/dev-container-docker-compose-volume-or-bind to check for that.

**Reply by jehon** [reply]
Am I a rare user of this use case? Should this not be implemented more widely?

**Reply by chrmarti** [reply]
I agree, we could try to do this automatically. Could you open a feature request at https://github.com/microsoft/vscode-remote-release/issues/new?assignees=&labels=&template=feature_request.md?

**Reply by jehon** [reply]
Posted here https://github.com/microsoft/vscode-remote-release/issues/7924 for anyone coming here...
